### DISCUSSION TITLE
Issue when following Music Store App Guide

### DISCUSSION THREAD

**Discussion by MicahWade** [AvaloniaUI/Avalonia#16513] [Q&A] [answered]
### Describe the bug

When following the Guide and i get to Open a Dialog Vscode with the extension flags `        this.WhenActivated(d => d(ViewModel!.ShowDialog.RegisterHandler(DoShowDialogAsync)));` for DoShowDialogAsync underlined with read This is what is said Argument 1: cannot convert from 'method group' to 'System.Action<ReactiveUI.IInteractionContext<MusicApp.ViewModels.MusicStoreViewModel, MusicApp.ViewModels.AlbumViewModel? and getting some error when trying to run or build

### To Reproduce

Copy the code from the Guide on the website https://docs.avaloniaui.net/docs/tutorials/music-store-app/opening-a-dialog

### Avalonia version

11.1.0

### OS

Windows

### Additional context

Sorry if it is not written well.

**Comment by thevortexcloud** [comment]
Can you share your full code for your code behind? It's likely a signature problem.

**Comment by MicahWade** [comment]
Here is the Code it is the same as in the Guide
```
using Avalonia.ReactiveUI;
using ReactiveUI;
using MusicApp.ViewModels;
using System.Threading.Tasks;

namespace MusicApp.Views;

public partial class MainWindow : ReactiveWindow<MainWindowViewModel>
{
    public MainWindow()
    {
        InitializeComponent();
        this.WhenActivated(d => d(ViewModel!.ShowDialog.RegisterHandler(DoShowDialogAsync)));
    }
    private async Task DoShowDialogAsync(InteractionContext<MusicStoreViewModel, 
                                            AlbumViewModel?> interaction)
    {
        var dialog = new MusicStoreWindow();
        dialog.DataContext = interaction.Input;
        var result = await dialog.ShowDialog<AlbumViewModel?>(this);
        interaction.SetOutput(result);
    }
}
```

**Comment by IanRawley** [comment]
How about the ViewModel, specifically the definition of ShowDialog?

**Comment by maxkatz6** [comment]
Add "using System;"
It's a common mistake with ReactiveUI.

**Comment by MicahWade** [comment]
Added using System But Still get this error
>  error CS1503: Argument 1: cannot convert from 'method group' to 'System.Action<ReactiveUI.IInteractionContext<MusicApp.ViewModels.MusicStor
> eViewModel, MusicApp.ViewModels.AlbumViewModel?>>

I have Goan through the guide again but got the same error. Think this may be because the Guide is out dated?

**Comment by Georapella** [comment, marked as answer]
I'm going through this tutorial too and am having the same problem.

I think I found a solution though...

1. Add **.UseReactiveUI()** to **AppBuilder** in _Program.cs_.
2. Make sure **ViewModelBase** inherits from **ReactiveObject** in _ViewModels/ViewModelBase.cs_
3. Change **InteractionContext** to **IInteractionContext** for **DoShowDialogAsync** in _Views/MainWindow.axaml.cs_.

**Reply by MicahWade** [reply]
Thank you so much.

**Reply by MelkOuss** [reply]
thanks @Georapella, 

also just in case for future developers, the package may be missing in the project file (as it was for me). You may need to verify that it was referenced as it should be by adding it to the .csproj file: 

`<ItemGroup>
  <PackageReference Include="Avalonia.ReactiveUI" Version="11.2.3" />
</ItemGroup>`

**Reply by stu640978** [reply]
Thanks so much.
You saved my time :)

**Comment by MicahWade** [comment]
This Should be Changed in the Documentation
